addons/point_of_sale/static/src/app/store/make_awaitable_dialog.js

```diff
@@ -18,6 +18,25 @@ export function makeAwaitable(dialog, comp, props, options) {
     });
 }
 
+export function makeActionAwaitable(action, config, additionalArgs) {
+    const props = additionalArgs?.props || {};
+    delete additionalArgs.props;
+    return new Promise((resolve) => {
+        action.doAction(config, {
+            ...additionalArgs,
+            props: {
+                ...props,
+                onSave: (record) => {
+                    action.doAction({
+                        type: "ir.actions.act_window_close",
+                    });
+                    resolve(record);
+                },
+            },
+        });
+    });
+}
```

Comment:
Can you find a way to not change the arguments? Maybe just passing the props is enough
```suggestion
export function makeActionAwaitable(action, config, props) {

    return new Promise((resolve) => {
        action.doAction(config, {
            props: {
                ...props,
                onSave: (record) => {
                    action.doAction({
                        type: "ir.actions.act_window_close",
                    });
                    resolve(record);
                },
            },
        });
    });
}
```

Reply:
I need to override onSave anyway to resovle, and you get the value of onSave as the result of the promise. For the rest is standard use

Reply:
I just stop to use extra var, but its the same thing.

Reply:
I agree with the original. doAction's 2nd args may contain other field that `props`.